packages/core-api/src/apis/implementations/StorageApi/WebStorage.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { StorageApi, ObservableMessage } from '../../definitions';
+import { Observable } from '../../../types';
+import ObservableImpl from 'zen-observable';
+
+export class WebStorage implements StorageApi {
+  private readonly observable = new ObservableImpl<ObservableMessage>(() => {});
+  get<T>(key: string): T | undefined {
+    try {
+      const storage = JSON.parse(localStorage.getItem(key)!);
+      return storage ?? undefined;
+    } catch (e) {
+      window.console.error(
+        `Error when parsing JSON config from storage for: ${key}`,
+        e,
+      );
+    }
+
+    return undefined;
+  }
+
+  async set<T>(key: string, data: T): Promise<void> {
+    localStorage.setItem(key, JSON.stringify(data, null, 2));
+  }
+
+  async remove(key: string): Promise<void> {
+    localStorage.removeItem(key);
+  }
+
+  observe$<T>(key: string): Observable<T> {
+    return this.observable.filter(
+      ({ key: messageKey }) => messageKey === key,
+    ) as Observable<T>;
+  }
```

Comment:
@Rugvip - not really sure if this is a good idea or wether we should just use the `PublishSubject` and let the the consumer deal with filtering out keys they care about in the store? Otherwise I could add the `filter` method to the `PublishSubject` too which I'd be interested to get opinions about? Whats your thoughts?

Reply:
For now - I just created a simple `Observer` in the `storage-api` as I wanna try and get this merged to so I can build upon it for the react hook.

Reply:
I think this is better than letting the consumer filter out the keys, since that's possible to go spectacularly wrong :p Also think it would be fine to add both filter and map to our subjects.